dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/trackedentity/hibernate/HibernateTrackedEntityInstanceStore.java

```diff
@@ -484,11 +486,55 @@ private String getQuerySelect( TrackedEntityInstanceQueryParams params, boolean
         {
             select.append( ", TEI.trackedentityinstanceid AS teiid" );
         }
+
+        if ( params.getOrders() != null )
+        {
+            select.append( getOrderByQuery( params, select ) );
+        }
+
         select.append( SPACE );
 
         return select.toString();
     }
 
+    private String getOrderByQuery( TrackedEntityInstanceQueryParams params, StringBuilder select )
+    {
+        StringBuilder orderQuery = new StringBuilder();
+
+        for ( OrderParam orderParam : params.getOrders() )
+        {
+            if ( isStaticColumn( orderParam.getField() ) )
+            {
+                if ( ENROLLED_AT.getPropName().equalsIgnoreCase( orderParam.getField() ) )
+                {
+                    //In the main query, we need to use the alias "tei" to fetch the value of the field enrolledAt
+                    orderQuery
+                        .append( addStaticColumn( orderParam.getField(), select.toString() ).replace( "pi", "tei" ) );
+                }
+                else
+                {
+                    orderQuery.append( addStaticColumn( orderParam.getField(), select.toString() ) );
+                }
+            }
+        }
+
+        select.append( SPACE );
```

Comment:
Is this needed? This is also done right after the method returns in https://github.com/dhis2/dhis2-core/pull/12502/files#diff-f960125887353fe285946328173a4e612632e40c0bca283e9d30c824348764dbR497

If not you could pass in the select as `String` and save making two calls to build the string. This would also make it clear that you are not mutating the select.